tests/cli/v012/test_init.py

```diff
@@ -163,84 +164,88 @@ def test_cli_init_connection_string_non_working_db_connection_instructs_user_and
 ):
     root_dir = tmp_path_factory.mktemp("bad_con_string_test")
     root_dir = str(root_dir)
-    os.chdir(root_dir)
 
-    runner = CliRunner(mix_stderr=False)
-    result = runner.invoke(
-        cli,
-        ["init"],
-        input="\n\n2\n6\nmy_db\nsqlite:////subfolder_thats_not_real/not_a_real.db\n\nn\n",
-        catch_exceptions=False,
-    )
-    stdout = result.output
-    assert mock_webbrowser.call_count == 0
-
-    assert "Always know what to expect from your data" in stdout
-    assert "What data would you like Great Expectations to connect to" in stdout
-    assert "Which database backend are you using" in stdout
-    assert "What is the url/connection string for the sqlalchemy connection" in stdout
-    assert "Give your new Datasource a short name" in stdout
-    assert "Attempting to connect to your database. This may take a moment" in stdout
-    assert "Cannot connect to the database" in stdout
-
-    assert "Profiling" not in stdout
-    assert "Building" not in stdout
-    assert "Data Docs" not in stdout
-    assert "Great Expectations is now set up" not in stdout
-
-    assert result.exit_code == 1
-
-    ge_dir = os.path.join(root_dir, DataContext.GE_DIR)
-    assert os.path.isdir(ge_dir)
-    config_path = os.path.join(ge_dir, DataContext.GE_YML)
-    assert os.path.isfile(config_path)
-
-    config = yaml.load(open(config_path))
-    assert config["datasources"] == {
-        "my_db": {
-            "data_asset_type": {
-                "module_name": None,
-                "class_name": "SqlAlchemyDataset",
-            },
-            "credentials": "${my_db}",
-            "class_name": "SqlAlchemyDatasource",
-            "module_name": "great_expectations.datasource",
+    with set_directory(root_dir):
```

Comment:
These diffs are huge but the only change here is this context manager (defined in `test_utils.py`).

We want to `cd` into our test environment and `cd` out upon test completion; rather than having to manually remember to `cd` back, we use this new context manager to do it for us.

Simply put, it does the `cd` and then reverts itself once you leave the `with` statement's scope.

Reply:
Thank you @cdkini this blurb is really helpful